testsuite/MDAnalysisTests/lib/test_util.py

```diff
@@ -74,165 +63,163 @@ class TestStringFunctions(object):
         ("M1:CA", ("MET", 1, "CA")),
     ]
 
-    def test_parse_residue(self):
-        for rstring, residue in self.residues:
-            yield check_parse_residue, rstring, residue
+    @pytest.mark.parametrize('rstring, residue', residues)
+    def test_parse_residue(self, rstring, residue):
+        assert util.parse_residue(rstring) == residue
 
     def test_parse_residue_VE(self):
-        assert_raises(ValueError, util.parse_residue, 'ZZZ')
+        with pytest.raises(ValueError):
+            util.parse_residue('ZZZ')
+
+    def check_convert_aa_3to1(self, resname3, resname1):
+        assert util.convert_aa_code(resname3) == resname1
 
+    # TODO: How do I fix this? Function call in parametrize?
     def test_convert_aa_code_long(self):
         for resname1, strings in self.aa:
             for resname3 in strings:
-                yield check_convert_aa_3to1, resname3, resname1
+                yield self.check_convert_aa_3to1, resname3, resname1
```

Comment:
doesn't the yield still raise a internal pytest warning? We should remove all the yields.